clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/ops/DeleteGoogleAutoscalingPolicyAtomicOperation.groovy

```diff
@@ -40,10 +42,10 @@ class DeleteGoogleAutoscalingPolicyAtomicOperation extends GoogleAtomicOperation
   private final DeleteGoogleAutoscalingPolicyDescription description
 
   @Autowired
-  private GoogleClusterProvider googleClusterProvider
+  @VisibleForTesting GoogleClusterProvider googleClusterProvider
```

Comment:
I'd prefer refactoring this to use constructor injection.  With that I think it'll be possible to keep these members private since DeleteGoogleAutoscalingPolicyAtomicOperationUnitSpec can pass mocks in to the constructor.

Reply:
Update the code with constructor-based DI. All tests belonging to `DeleteGoogleAutoscalingPolicyAtomicOperationUnitSpec` are passed, under upgraded env. as well.